### DISCUSSION TITLE
Welcome to Ultimate-Proxy Discussions ! 👋

### DISCUSSION THREAD

**Discussion by romslf** [romslf/Ultimate-Proxy#2] [General]
## 👋 Welcome!
  
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by romslf** [comment]
Hello, I made some updates that I did not pushed online for now, but I
would love to help you, do you have any questions or suggestions?

**Comment by romslf** [comment]
I began to add a Rest API, a web user interface, Beam support and more.
Let me know if you would like to try it.
I need to double check everything before releasing it, a beta tester
already tried it and did not encounter any bug.
But I need time to work on it.

**Reply by Bitofsin** [reply]
I'm willing to test out any of your new releases if you need.

 I was wondering if it would be possible to add a time-based pool switching like a Dev fee. Where you can mine for x percent to one pool and switch to another for the other percentage of time.

**Reply by romslf** [reply]
I will look into this today, if you wish you can join the discord server so we can communicate easily

https://discord.gg/GyP65USN

**Reply by Bitofsin** [reply]
Just joined

**Comment by i8088** [comment]
Hello. I' like try this but I want to mine solo eth 2miners  non-stop and no profit switch. How hot set it ? 
What is best settings for Eth solo on 2miners ? 
And
is it importand value that  "NodeGetWorkIntervalMs": 500,     ?

**Reply by romslf** [reply]
Hey 👋

Thank you for your interest in ultimate proxy !

I responded by email but I'm not sure if it worked as the reply should have showed here too.

If you want to solo mine using a pool the « NodeGetWorkIntervalMs » is not used, it is only used if you are mining to your own full ETH node. So for mining solo using 2miners, just point it to the 2miners solo address and it should work as expected.

Fell free to join our discord for a quicker and simpler support, you can find it here: 
https://discord.gg/zWsTZXBYYq

Good luck, and happy mining !

**Reply by romslf** [reply]
This config should work 

```

{
	"allowedAddresses": [
		"0.0.0.0"	// This allow every IP to connect to proxy, set only needed IPs if more security is needed
	],
	"poolList": [
		{
			"address": "solo-eth.2miners.com",
			"port": 16060,
			"ssl": true
		},
		{
			"address": "solo-eth.2miners.com",
			"port": 6060
		}
	],
        "Protocol": "Stratum", // The mining protocol used (Ethproxy/Stratum/Nicehash)
	"Coin": "ETH",	// The coin you want to mine
	"Wallet": "YOUR WALLET HERE",	// Your mining wallet
	"Worker": "UltimateProxy",	// Proxy worker name
	"Password": "x",	// Proxy password
	"RatioWindowTimeHours": 1,
	"ProxyPort": 4444,	// Proxy port
	"ProxyCert": "",	// Set it if you want your workers to connect to proxy using SSL (See "Docs" folder create a .pfx file)
	"PrintStats": true,	// Display workers/pools stats
	"StatsIntervalSeconds": 60,	// Delay between PrintStats
	"NodeGetWorkIntervalMs": 500,	// Delay between node solo getWork requests
        "StaleSharesWindow": 10,
        "ForceWorkersReconnect": false
}

```

_I writed it on phone, let me know if I made a mistake_

I will update the git documentation in the coming days as it is outdated.

**Comment by romslf** [comment]
Hey, thank you for your interest in ultimate proxy !

If you want to solo mine using a pool the « NodeGetWorkIntervalMs » is not
used, it it used only if you use your own ETH full node. So for mining solo
using 2miners, just point it to the 2miners solo address and it should work
as expected.

Fell free to join our discord for a quicker and simpler support, you can
find it here:
https://discord.gg/zWsTZXBYYq

Good luck, and happy mining !
